src/main/java/com/microsoft/azure/spring/data/documentdb/common/PropertyLoader.java

```diff
@@ -5,40 +5,55 @@
  */
 package com.microsoft.azure.spring.data.documentdb.common;
 
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Properties;
 
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
 public class PropertyLoader {
+
     private static final String PROJECT_PROPERTY_FILE = "/META-INF/project.properties";
+    private static final String APPLICATION_PROPERTY_FILE = "/application.properties";
+    private static final String APPLICATION_YML_FILE = "/application.yml";
 
-    private PropertyLoader() {
-        // Hide the implicit public one
+    public static String getProjectVersion() {
+        return getPropertyByName("project.version", PROJECT_PROPERTY_FILE);
     }
 
-    public static String getProjectVersion() {
-        String version = "unknown";
-        InputStream inputStream = null;
-        try {
-            inputStream = PropertyLoader.class.getResourceAsStream(PROJECT_PROPERTY_FILE);
-            if (inputStream != null) {
-                final Properties properties = new Properties();
-                properties.load(inputStream);
+    public static String getApplicationTelemetryAllowed() {
```

Comment:
how about parse to bool directly here

Reply:
you mean just return boolean, and put the string handling inside propertyloader ?